Chess/Assets/Player.cs

```diff
@@ -206,18 +242,19 @@ private Task SubscribeToPlayerMessages()
 
     public void PlayerInteract(InputAction.CallbackContext context)
     {
-        if (!context.performed && _currentLobby != null) return;
+        if (!context.performed && _currentSession != null) return;
         var mousePosition = Mouse.current.position.ReadValue();
         var rayOrigin = playerCamera.ScreenPointToRay(mousePosition);
         if (Physics.Raycast(rayOrigin, out var hitInfo))
         {
-            // TODO: check if a piece is selected and another piece is clicked - is that a move?
-            if (hitInfo.transform.gameObject.name == "Board")
+            var gameObject = hitInfo.transform.gameObject;
+            if (hitInfo.transform.gameObject.name == "Board"
```

Comment:
While we're making changes here, could it be better to store a reference to the instance of the board gameObject in the player script, then evaluate an expression like this

```
hitInfo.transform.gameObject == this._board
```

Reply:
That does sound better. Also we could probably use tags for the pieces rather than checking their names for Light/Dark prefixes